source/core/src/main/com/csse3200/game/components/tasks/UseItemTask.java

```diff
@@ -0,0 +1,86 @@
+package com.csse3200.game.components.tasks;
+
+import com.csse3200.game.ai.tasks.DefaultTask;
+import com.csse3200.game.ai.tasks.PriorityTask;
+import com.csse3200.game.components.player.PlayerInventoryDisplay;
+import com.csse3200.game.entities.Entity;
+import com.csse3200.game.events.EventHandler;
+import com.csse3200.game.inventory.items.AbstractItem;
+import com.csse3200.game.inventory.items.ItemUsageContext;
+import com.csse3200.game.inventory.items.potions.AbstractPotion;
+import com.csse3200.game.inventory.items.potions.AttackPotion;
+import com.csse3200.game.inventory.items.potions.DefensePotion;
+import com.csse3200.game.inventory.items.potions.SpeedPotion;
+import com.csse3200.game.services.GameTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+public class UseItemTask extends DefaultTask implements PriorityTask {
+    private final Entity target;
+    private AbstractItem potion;
+    private static final Logger logger = LoggerFactory.getLogger(UseItemTask.class);
+    private final int priority;
+    private ItemUsageContext context;
+    private static final long DURATION = 120000;
+
+    public UseItemTask(Entity target, int priority, AbstractItem potion, ItemUsageContext context) {
+        this.target = target;
+        this.priority = priority;
+        this.potion = potion;
+        this.context = context;
+    }
+
+    /**
+     * Starts the task by adding an event listener for the "useItem" event.
+     * The event is triggered when the Player presses P.
+     */
+    @Override
+    public void start() {
+        super.start();
+        this.target.getEvents().addListener("itemUsed", this::activePotion);
+    }
+
+    /**
+     * Updates the task each frame. It checks the players proximity to the item and displays
+     * an overlay if the player is near. If the play moves away, the overlay is removed.
+     */
+    @Override
+    public void update() {
+        super.update();
+        if(activePotion() && this.potion instanceof DefensePotion) {
+            if (((DefensePotion) this.potion).isExpired(this.context)) { //check if the item is in use
```

Comment:
Rather than having so much code duplication, just create an item class which handles having an item be used over a period of time and has abstract methods of isExpired and update - then this class can take in a conrete subclass of that rather than AbstractItem. Probably get rid of AbstractPotion as well since it doesn't do anything rn.